language-support/hs/bindings/src/LedgerIdentity.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs     #-}
+
+module LedgerIdentity(
+  LedgerId,
+  ledgerId,
+  Port(..),
+  )
+where
+
+import qualified Data.Map           as Map
+import           Data.String        (fromString)
+import qualified Data.Text.Lazy     as Text
+
+import Network.GRPC.HighLevel.Generated (
+  withGRPCClient,
+  ClientRequest(
+      --ClientReaderRequest
+      ClientNormalRequest
+      ),
+  ClientResult(
+        --ClientReaderResponse,
+      ClientNormalResponse,
+      ClientErrorResponse
+      ),
+  MetadataMap(..),
+  GRPCMethodType(Normal
+                 --ServerStreaming,
+                 ),
+  ClientConfig(..),
+  Port(..),
+  Host(..),
+  )
+
+import Network.GRPC.HighLevel.Client(
+  TimeoutSeconds,
+  )
+
+import Com.Digitalasset.Ledger.Api.V1.TraceContext(TraceContext)
+import Com.Digitalasset.Ledger.Api.V1.LedgerIdentityService as LIS
+    
+data LedgerId = LedgerId String deriving Show
```

Comment:
Given that you get it back as a `Text` below, why not make this a newtype around `Text`?

Reply:
ok